### ISSUE TITLE
Chrome extension keeps saying "Generating Encryption Keys..."

### ISSUE THREAD

**Issue by tiede** [cryptocat/cryptocat#700]
The chrome extension never completes and just keeps saying "Generating Encryption Keys...". 

It's a problem on Chrome version 39.0.2171.95 and Chrome Canary version 41.0.2259.0 canary. I tried the FireFox extension and it works flawlessly.

**Comment by mykhael** [comment]
@tiede I'm experimenting this same issue, this seems to be an issue with the desktop version of chrome not processing the ssl certification key from https://crypto.cat. I've tested the site connection via my mobile device and the connection seemed to be perfectly fine. I'll be testing this a little more to try to discover some sort of fix for this, but till then try to clear your browser's cache.

**Comment by collina** [comment]
![screen shot 2015-01-06 at 8 13 49 am](https://cloud.githubusercontent.com/assets/582005/5628990/3d2d7960-957c-11e4-8d35-089ca8872264.png)

**Comment by mykhael** [comment]
@collina This is exactly the issue, at this point I have no idea what could be happening

**Comment by jkiv** [comment]
True, Chrome doesn't like the SSL/TLS certificate for `https://crypto.cat` but does like the certificate for `https://www.crypto.cat`.

Changing `Cryptocat.xmpp.defaultRelay` from `https://crypto.cat` to `https://www.crypto.cat` fixed this particular issue for me: https://github.com/cryptocat/cryptocat/blob/32fd02f8d899e219a004281eb0ce364cb52dd62a/src/core/js/etc/xmpp.js#L10

I'm not sure how other instances/uses of `https://crypto.cat` affect the behaviour yet, as I haven't looked much deeper. There appears to be some other issues too after once my room comes up (not sure if this is normal):
![image](https://cloud.githubusercontent.com/assets/226881/5651341/a03d0220-9674-11e4-8e54-f256cf8c5470.png)

**Comment by mykhael** [comment]
@tiede @jkiv you are right, the browser does accept `https://www.crypto.cat` and I'll test this issue. If this works, I'd suggest that we either close this or mark as bug.

**Comment by mykhael** [comment]
@tiede for now just open cryptocat and change `https://crypto.cat/http-bind` to `https://www.crypto.cat/http-bind`, this will allow you to connect and now please may you mark this issue as `bug`

**Comment by mykhael** [comment]
@jkiv also the input that is logged in the console is warned in the comment syntax as `// Fires the XHR request -- may be invoked immediately` and `// or on a gradually expanding retry window for reconnects`  so it's going to happen at some point but isn't anything major.

**Comment by tiede** [comment]
Works fine with www.crypto.cat. 

The problem is that the certificate for crypto.cat is pinned in the Chrome browser: 

chrome://net-internals/#hsts and query for crypto.cat.

It looks like the crypto.cat certificate was renewed on december 24, 2014 and the signature of the new certificate has properly not been updated in Chrome.

BTW: I do not know how to mark it as a bug - dont think I have access to do it.

**Comment by mykhael** [comment]
@tiede I noticed that the certificate was renewed as well and sometimes issues happen when re-implementing new certificates so it may have affected the crypto.cat source it self, but since we don't have the access to mark this as bug I guess that we'll have to wait for a contributer

**Comment by signaleleven** [comment]
I would not classify this a bug in crypto.cat rather than a problem with the hardcoded pinning in Chrome.
 
The pinning for crypto.cat and other sites is hardcoded in Chrom[e|ium], see [here](https://code.google.com/p/chromium/codesearch#chromium/src/net/http/transport_security_state_static.json)

What needs to happen is for someone to update the hash of the key chrome expects to see for crypto.cat and consider it valid for subdomains (because the fact that we can access crypto.cat with www is wrong, considering that it is using the same cert and key that chrome considers wrong for crypto.cat).

I have a very rough grasp of HSTS so I am not 100% confident that what I said is right, but [this issue](https://code.google.com/p/chromium/issues/detail?id=446240&q=crypto.cat&colspec=ID%20Pri%20M%20Week%20ReleaseBlock%20Cr%20Status%20Owner%20Summary%20OS%20Modified) on the chromium project seems just about this!

**Comment by tiede** [comment]
@signaleleven You are right. The fix of the issue you refer to removes the pining of the crypto.cat certificate, so that Chrome no longer checks the hash of the crypto.cat certificate. I can close this issue since it will be fixed in Chrome 40. Another fix could be to change the http-bind for the chrome app to www.crypto.cat and release a new version?
